src/Nocarrier/HalLink.php

```diff
@@ -80,6 +80,31 @@ public function getAttributes()
     }
 
     /**
+     * Set a particular attribute on this link.
+     *
+     * @param string $name
+     *   Name of the attribute to set.
+     * @param mixed $value
+     *   Value to set the attribute to.
+     * @return void
+     */
+    public function setAttribute($name, $value)
+    {
+        $this->attributes[$name] = $value;
+    }
+
+    /**
+     * Set the attributes on this link.
+     *
+     * @param array $attributes
+     *   Attibutes to set on this link.
+     */
+    public function setAttributes(array $attributes)
```

Comment:
Similar to my note above, I think this is asking for problems. I would make people use the above `setAttribute` method and be explicit about whether they are intentionally overwriting something.